test/sequential/sequencerStaking.rules.test.ts

```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
+      ),
+    );
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("false");
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        sequencer.keyRingPair.address,
+        new L2Update(api)
+          .withCancelResolution(txIndex, 1, true)
+          .on(chain)
+          .buildUnsafe(),
+      ),
+    ).then(async (events) => {
+      expectMGAExtrinsicSuDidSuccess(events);
+    });
+    const event = await waitForEvents(api, "rolldown.RequestProcessedOnL2", 40);
+    const err = await getEventError(event);
+    expect(err).toEqual("WrongCancelRequestId");
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("true");
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationMembers[0],
+        Maintenance.switchMaintenanceModeOff(),
+      ),
+    );
+  });
+
+  it("When a cancel resolution fail, the whole update wont be stored", async () => {
+    let maintenanceStatus: any;
+    const foundationMembers = await FoundationMembers.getFoundationMembers();
+    const chain = "Ethereum";
+    const sequencer = await setupASequencer(chain);
+    const api = getApi();
+    const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationMembers[0],
+        Maintenance.switchMaintenanceModeOff(),
+      ),
+    );
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("false");
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    const update = new L2Update(api)
+      .withDeposit(
+        txIndex,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_HUNDRED,
+      )
+      .withCancelResolution(txIndex, 1, true)
+      .on(chain)
+      .buildUnsafe();
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update),
+    ).then(async (events) => {
+      expectMGAExtrinsicSuDidSuccess(events);
+    });
+    const event = await waitForEvents(api, "rolldown.RequestProcessedOnL2", 40);
+    const depositErr = await getEventError(event);
+    const resolutionErr = await getEventError(event, 1);
+    expect(depositErr).toEqual(undefined);
+    expect(resolutionErr).toEqual("WrongCancelRequestId");
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("true");
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationMembers[0],
+        Maintenance.switchMaintenanceModeOff(),
+      ),
+    );
+  });
```

Comment:
I think u forgot to check the status of the deposit, did it work ? Can u ensure if it did or did not? 
Why reusing the same txIndex for both deposit and cancel resolution?